utils/zerovec/src/hashmap/mod.rs

```diff
@@ -207,24 +61,19 @@ where
     K: ZeroMapKV<'a> + ?Sized + Hash + Eq,
     V: ZeroMapKV<'a> + ?Sized,
 {
-    /// Given a [`key`] return the probable index of the key or [`None`] if the key is guaranteed to be absent.
+    /// Given a `key` return the probable index of the key or [`None`] if the key is guaranteed to be absent.
     /// Another check to determine if the key matches the one at the probable index is required.
     #[inline]
     pub fn index<A>(&self, key: &A) -> Option<usize>
     where
-        A: Borrow<K>,
+        A: Borrow<K> + ?Sized,
     {
-        let (g, f0, f1) = compute_hash(key, self.displacements.len());
+        let hash = compute_hash(key.borrow());
+        let (g, f0, f1) = split_hash64(hash, self.len() as u32);
```

Comment:
`m` and `g` should always be `usize`, because they're a length and an index. Casting them around is confusing to the compiler (that might miss optimisations) and the reader.

Reply:
Changed `m` to always be `usize`.